multi-buffer-search/src/services/searchService.ts

```
@@ -0,0 +1,158 @@
+          matches.push(result as vscode.TextSearchMatch)
+        }
+      }
+    )
+
+    const matchesByFile = new Map<string, vscode.TextSearchMatch[]>()
+    for (const match of matches) {
+      const key = match.uri.toString()
+      if (!matchesByFile.has(key)) {
+        matchesByFile.set(key, [])
+      }
+      matchesByFile.get(key)!.push(match)
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)